README.adoc

```diff
@@ -0,0 +1,478 @@
+The documentation itself is located in a dedicated subproject in each software branch (i.e., the docs are stored alongside the code).
+Software branches, also referred to as release line branches or content branches, follow the pattern `major.minor.x` (e.g., 6.0.x), or `main` for the latest release line.
+Software tags follow the pattern `major.minor.patch` (e.g., 6.0.1).
+The latest version of the docs for each release line is always sourced from a tag.
+The release line branches host the prerelease materials for the next version.
+
+=== Builds
+
+The build for the production site is stored at the root of the playbook branch.
+This branch also holds the search crawler configuration and runner for the production site.
+This build is only needed for building the production site (or for developing and testing the docs build itself).
+The build for each content branch is located in a subproject, typically the folder named _docs_.
+This is the build that authors use to preview the docs for a single version when writing content.
+
+Regardless of how the docs site is built, Antora is configured to run a separate command to compute the version of the docs per git reference.
+This command is run by an Antora extension named Antora Collector.
+Using Collector allows the version of the docs to be maintained centrally in the _gradle.properties_ file in each git reference.
+The command also populates a collection of AsciiDoc attributes that provide access to software versions and resource URLs.
+_The docs build will not work without Collector._
+
+=== UI
+
+The UI is developed in a separate project named https://github.com/spring-io/antora-ui-spring[antora-ui-spring].
+That project generates and publishes a UI bundle, which the docs build refers to using its public bundle URL.
+There is one UI for all versions of the documentation.
+
+The UI only shows the latest version in each release line.
+In order to access other versions that are published, the URL must be known in advance.
+
+[#usage]
+== Usage
+
+To prepare your system for building the documentation, <<prerequisites,install the prerequisites>>, then <<prepare-workspace,prepare your workspace>>.
+Then you can build the documentation in either the main branch or 6.0.x branch.
+
+.Branch checkout instead of worktrees
+[NOTE]
+====
+If you prefer to set up your workspace without worktrees, complete the steps in <<prerequisites,Prerequisites>> and clone the project repository onto your computer.
+Then, check out the desired branch and follow the instructions in each section starting from the `sdk env || sdk env install` step.
+====
+
+[#prerequisites]
+=== Prerequisites (everyone)
+
+These instructions assume you already have basic tools on your system, including bash, zip, unzip, git, and curl.
+In addition to these basic tools, you need https://sdkman.io/install[SDKMAN!] installed so that the correct JDK is set for each branch.
```

Comment:
Some commands down the readme use `gh`, which is not mentioned here. I will take care of updating the list.